projects/frontend/cicd/.gitlab-ci.yml

```diff
@@ -1,6 +1,59 @@
 # Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
+.frontend_shared_components_changes: &frontend_shared_components_locations
+  - "projects/frontend/cicd/**/*"
+  - "projects/frontend/shared-components/**/*"
+
+.frontend_data_pipelines_changes: &frontend_data_pipelines_locations
+  - "projects/frontend/cicd/**/*"
+  - "projects/frontend/data-pipelines/**/*"
+
+frontend-data-pipelines-build:
+  image: "versatiledatakit/vdk-cicd-base-gui:2.0.0"
+  stage: build
+  before_script:
+    - cd projects/frontend/
+  script:
+    - ./cicd/install_data_pipelines.sh
+  coverage: /^TOTAL\s+\d+\s+\d+\s+(\d+\%)$/
+  retry: !reference [.retry, retry_options]
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "main" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
+      changes: *frontend_shared_components_locations
+    - if: '$CI_COMMIT_BRANCH == "main" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
+      changes: *frontend_data_pipelines_locations
+  artifacts:
+    when: always
+    reports:
+      junit:
+        - projects/frontend/data-pipelines/gui/reports/test-results/data-pipelines-ui/*.xml
+        - projects/frontend/data-pipelines/gui/reports/test-results/data-pipelines-lib/*.xml
+    paths:
+      - projects/frontend/data-pipelines/gui/package-lock.json
+    expire_in: 1 week
+
+frontend-data-pipelines-release:
+  image: "versatiledatakit/vdk-cicd-base-gui:2.0.0"
+  stage: release
+  before_script:
+    - cd projects/frontend/
+  script:
+    - ./cicd/publish_package_npm.sh data-pipelines
+  retry: !reference [.retry, retry_options]
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+      when: never
+    # TODO: cleanup external_pull_request_event before merge; for dev verification purposes only
+    - if: '$CI_COMMIT_BRANCH == "main" || $CI_PIPELINE_SOURCE == "external_pull_request_event"'
```

Comment:
will cleanup this todo before merge; 
external_pull_request_event used to replay the entire workflow in advance for review